selfdrive/car/docs.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+import jinja2
+import os
+from typing import Dict, Iterator, List, Tuple
+
+from common.basedir import BASEDIR
+from common.params import Params
+from selfdrive.car.docs_definitions import CarFootnote, Column, Star, Tier
+from selfdrive.car.car_helpers import interfaces, get_interface_attr
+from selfdrive.car.hyundai.radar_interface import RADAR_START_ADDR as HKG_RADAR_START_ADDR
+from selfdrive.car.tests.routes import non_tested_cars
+
+
+def get_all_footnotes() -> Dict[CarFootnote, int]:
+  all_footnotes = {}
+  i = 1
+  for _, footnotes in get_interface_attr("FOOTNOTES").items():
+    if footnotes is not None:
+      for footnote in footnotes.values():
+        all_footnotes[footnote] = i
+        i += 1
+  return all_footnotes
+
+
+ALL_FOOTNOTES: Dict[CarFootnote, int] = get_all_footnotes()
+CARS_MD_OUT = os.path.join(BASEDIR, "docs", "CARS.md")
+CARS_MD_TEMPLATE = os.path.join(BASEDIR, "selfdrive", "car", "CARS_template.md")
+
+
+def get_tier_car_rows() -> Iterator[Tuple[str, List[str]]]:
+  tier_car_rows: Dict[Tier, list] = {tier: [] for tier in Tier}
+
+  # TODO: Remove Hyundai exceptions once full long support is added
+  # Cars that can disable radar have openpilot longitudinal
+  Params().put_bool("DisableRadar", True)
+
+  for models in get_interface_attr("CAR_INFO").values():
+    for model, car_info in models.items():
+      # Hyundai exception: all have openpilot longitudinal
+      fingerprint = {0: {}, 1: {HKG_RADAR_START_ADDR: 8}, 2: {}, 3: {}}
```

Comment:
this isn't exactly right. they don't all use the same radar and we only parse the one the sonata uses currently